### PR TITLE
Clean up push rules management and fixes several issues

### PR DESCRIPTION
**Pull request by bmarty**
This PR fixes several issues:
- Clear cache was losing the push rules. Push rules are now retrieved from the sync, so are always up to date
- (Should fix) m.notice trigger notification (#238)
- simplify db entity by removing useless userId field
- Fix TU compilation error

It also introduce @UserId, to inject the userId as a String

### CODE REVIEW

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/sync/SyncResponseHandler.kt [hunk 1]
```diff
@@ -33,7 +33,7 @@ internal class SyncResponseHandler @Inject constructor(private val roomSyncHandl
                                                        private val cryptoService: DefaultCryptoService,
                                                        private val initialSyncProgressService: DefaultInitialSyncProgressService) {
 
-    fun handleResponse(syncResponse: SyncResponse, fromToken: String?, isCatchingUp: Boolean): Try<SyncResponse> {
+    suspend fun handleResponse(syncResponse: SyncResponse, fromToken: String?, isCatchingUp: Boolean): Try<SyncResponse> {
```
Comment (ganfra) [inline matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/sync/SyncResponseHandler.kt]: If it becomes suspendable, we should remove Try

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/algorithms/megolm/MXMegolmDecryptionFactory.kt [hunk 2]
```diff
@@ -16,18 +16,19 @@
...
 import im.vector.matrix.android.internal.crypto.actions.EnsureOlmSessionsForDevicesAction
 import im.vector.matrix.android.internal.crypto.actions.MessageEncrypter
 import im.vector.matrix.android.internal.crypto.store.IMXCryptoStore
 import im.vector.matrix.android.internal.crypto.tasks.SendToDeviceTask
+import im.vector.matrix.android.internal.di.UserId
 import im.vector.matrix.android.internal.util.MatrixCoroutineDispatchers
 import javax.inject.Inject
 
-internal class MXMegolmDecryptionFactory @Inject constructor(private val credentials: Credentials,
+internal class MXMegolmDecryptionFactory @Inject constructor(@UserId
```
Comment (ganfra) [inline matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/algorithms/megolm/MXMegolmDecryptionFactory.kt]: I prefer having annotations on same line for constructor properties...

Comment (bmarty) [reply matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/algorithms/megolm/MXMegolmDecryptionFactory.kt]: done

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/sync/SyncResponseHandler.kt (hunk 1 above)
Comment (bmarty) [reply matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/sync/SyncResponseHandler.kt]: ok, done. Also removing the return value, which is useless

**Pull request merged**